### PR TITLE
Add migration process to /dart-2

### PR DESCRIPTION
**Pull request by kwalrath**
Also rejiggered much of the page.
All this was spurred by the need to link to the dart2_fix tool.

Staged:
https://kw-staging-dartlang-2.firebaseapp.com/dart-2

### CODE REVIEW

File: src/dart-2.md [hunk 1]
```diff
@@ -5,23 +5,41 @@ title: Dart 2 Migration
...
+Here's the general process for migrating to Dart 2,
+from either Dart 1.x or an earlier version of Dart 2:
 
+1. Get an up-to-date version of the Flutter or Dart SDK.
+   * [Flutter SDK instructions][Flutter SDK install]
+   * [Dart SDK instructions][Dart SDK install] (VM or web)
+2. Upgrade the packages your app depends on.
+   * Flutter: [`flutter packages upgrade`][flutter package upgrade]
+   * Dart VM or web: [`pub upgrade`][pub upgrade]
+3. Run the [dart2_fix tool.][dart2_fix]
```
Comment (devoncarew) [inline src/dart-2.md]: nit: the period should be outside of the link

Comment (devoncarew) [reply src/dart-2.md]: and I see this in two links further down the page, so you may have a different opinion about where to put the periods :)

File: src/dart-2.md [hunk 2]
```diff
@@ -5,23 +5,41 @@ title: Dart 2 Migration
...
+{% comment %}
+TODO:
+- talk about library vs. app migration?
+- talk more about versioning?
+{% endcomment %}
+
+
+## Differences
 
 ### Language and libraries
```
Comment (chalin) [inline src/dart-2.md]: I missed this earlier: from a few lines below ...

    * [Email: Breaking Change: --preview-dart-2 turned on by default][Leaf's email]

The `--` gets converted to an m-dash when the markdown is process. Maybe code-format it?

* [Email: Breaking Change: `--preview-dart-2` turned on by default][Leaf's email]

[Leaf's email]: #

File: src/dart-2.md (hunk 1 above)
Comment (kwalrath) [reply src/dart-2.md]: Normally, I'd agree with you, but we have little icons for external links that look bad when they come before a period or comma.

File: src/dart-2.md (hunk 2 above)
Comment (kwalrath) [reply src/dart-2.md]: Oh! I hadn't noticed this. I'll fix that.

File: src/dart-2.md (hunk 1 above)
Comment (chalin) [reply src/dart-2.md]: (I've been considering writing a plugin to handle that.)

**Pull request merged**